Admin/Configuration/ApplicationConfiguration.php

```diff
@@ -0,0 +1,462 @@
+     */
+    protected $translationPattern;
+
+    public function __construct(array $applicationConfiguration = [], $locale)
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'enable_extra_configuration' => true,
+            'title' => '',
+            'description' => '',
+            'locale' => $locale,
+            'layout' => 'LAGAdminBundle::admin.layout.html.twig',
+            'block_template' => 'LAGAdminBundle:Form:fields.html.twig',
+            'bootstrap' => false,
+            'date_format' => 'd/m/Y',
+            'string_length' => 0,
+            'string_length_truncate' => '...',
+            'routing' => [
+                'url_pattern' => '/{admin}/{action}',
+                'name_pattern' => 'lag.admin.{admin}',
+            ],
+            'translation' => [
+                'enabled' => true,
+                'pattern' => 'lag.admin.{key}',
+            ],
+            'max_per_page' => 25,
+            'fields_mapping' => [
+                Field::TYPE_STRING => 'lag.admin.field.string',
+                Field::TYPE_ARRAY => 'LAG\AdminBundle\Admin\Render\ArrayRenderer',
+                Field::TYPE_LINK => 'LAG\AdminBundle\Admin\Render\LinkRenderer',
+                Field::TYPE_DATE => 'LAG\AdminBundle\Admin\Render\DateRenderer',
+                Field::TYPE_COUNT => 'LAG\AdminBundle\Admin\Render\CountRenderer',
+                Field::TYPE_ACTION => 'LAG\AdminBundle\Admin\Render\LinkRenderer',
+            ],
+        ]);
+        $resolver->setAllowedValues('enable_extra_configuration', [true, false]);
+        $applicationConfiguration = $resolver->resolve($applicationConfiguration);
+        // resolving routing options
+        $routingConfiguration = $applicationConfiguration['routing'];
+        $resolver->clear();
+        $resolver->setRequired([
+            'url_pattern',
+            'name_pattern',
+        ]);
+        $resolver->setNormalizer('url_pattern', function (Options $options, $value) {
+            if (strstr($value, '{admin}') === false) {
+                throw new InvalidOptionsException('Admin routing configuration url pattern should contains {admin} placeholder');
+            }
+            if (strstr($value, '{action}') === false) {
+                throw new InvalidOptionsException('Admin routing configuration url pattern should contains {action} placeholder');
+            }
+
+            return $value;
+        });
+        $resolver->setNormalizer('name_pattern', function (Options $options, $value) {
+            if (strstr($value, '{admin}') === false) {
+                throw new InvalidOptionsException('Admin routing configuration pattern name should contains {admin} placeholder');
+            }
+
+            return $value;
+        });
+        $routingConfiguration = $resolver->resolve($routingConfiguration);
+        // routing configuration
+        $this->routingUrlPattern = $routingConfiguration['url_pattern'];
```

Comment:
Maybe a dedicated class could be created to handle the route name and the url pattern at the same place.